tests/unit/test_util.py

```diff
@@ -15,186 +13,175 @@
-                "Positive electrode diffusivity [m2.s-1]": 6,
-            }
+def test_is_constant_and_can_evaluate():
+    symbol = pybamm.PrimaryBroadcast(0, "negative electrode")
+    assert not pybamm.is_constant_and_can_evaluate(symbol)
+    symbol = pybamm.StateVector(slice(0, 1))
+    assert not pybamm.is_constant_and_can_evaluate(symbol)
+    symbol = pybamm.Scalar(0)
+    assert True is pybamm.is_constant_and_can_evaluate(symbol)
+
+
+def test_fuzzy_dict():
+    d = pybamm.FuzzyDict(
+        {
+            "test": 1,
+            "test2": 2,
+            "SEI current": 3,
+            "Lithium plating current": 4,
+            "A dimensional variable [m]": 5,
+            "Positive electrode diffusivity [m2.s-1]": 6,
+        }
+    )
+    assert d["test"] == 1
+    with pytest.raises(KeyError, match="'test3' not found. Best matches are "):
+        d.__getitem__("test3")
+
+    with pytest.raises(KeyError, match="stoichiometry"):
+        d.__getitem__("Negative electrode SOC")
+
+    with pytest.raises(KeyError, match="dimensional version"):
+        d.__getitem__("A dimensional variable")
+
+    with pytest.raises(KeyError, match="open circuit voltage"):
+        d.__getitem__("Measured open circuit voltage [V]")
+
+    with pytest.raises(KeyError, match="Lower voltage"):
+        d.__getitem__("Open-circuit voltage at 0% SOC [V]")
+
+    with pytest.raises(KeyError, match="Upper voltage"):
+        d.__getitem__("Open-circuit voltage at 100% SOC [V]")
+
+    with pytest.warns(DeprecationWarning):
+        assert (
+            d["Positive electrode diffusivity [m2.s-1]"]
+            == d["Positive particle diffusivity [m2.s-1]"]
         )
-        self.assertEqual(d["test"], 1)
-        with self.assertRaisesRegex(KeyError, "'test3' not found. Best matches are "):
-            d.__getitem__("test3")
 
-        with self.assertRaisesRegex(KeyError, "stoichiometry"):
-            d.__getitem__("Negative electrode SOC")
 
-        with self.assertRaisesRegex(KeyError, "dimensional version"):
-            d.__getitem__("A dimensional variable")
-
-        with self.assertRaisesRegex(KeyError, "open circuit voltage"):
-            d.__getitem__("Measured open circuit voltage [V]")
-
-        with self.assertRaisesRegex(KeyError, "Lower voltage"):
-            d.__getitem__("Open-circuit voltage at 0% SOC [V]")
+def test_get_parameters_filepath():
+    tempfile_obj = tempfile.NamedTemporaryFile("w", dir=".")
+    assert pybamm.get_parameters_filepath(tempfile_obj.name) == tempfile_obj.name
+    tempfile_obj.close()
+
+    package_dir = os.path.join(pybamm.root_dir(), "pybamm")
+    tempfile_obj = tempfile.NamedTemporaryFile("w", dir=package_dir)
+    path = os.path.join(package_dir, tempfile_obj.name)
+    assert pybamm.get_parameters_filepath(tempfile_obj.name) == path
+    tempfile_obj.close()
+
+
+def test_is_jax_compatible():
+    if pybamm.have_jax():
+        compatible = pybamm.is_jax_compatible()
+        assert compatible
```

Comment:
Instead of having a conditional statement inside the test, you can move it up and use a decorator, i.e., `@pytest.mark.skipIf(pybamm.have_jax(), reason="The JAX solver is not installed")`